UnitexGramLab/gramlab-ide
# Feature/124

[feature #124] Allow to select an open graph to explore paths

## Description
 - Enable `Tools > Explore graph paths` if at least one graph is open
 - Change the type of inputGraphName from JTextField to JComboBox
 - Adapt the code that assumes inputGraphFrame is a JTextField
 - Make GraphPathFrame an observer of GraphFrame
 - Fill the inputGraphName with the list of opened GraphFrame

## Motivation and Context
[issue #124](https://github.com/UnitexGramLab/gramlab-ide/issues/124)

## How Has This Been Tested?

This new feature has been tested with the following scenarios:
1. Existing graph
   - Open a graph with the FSGraph menu
   - `FSGraph > Tools > Explore graph paths`
   - Click on run
   - We have the expected output
  
2. New graph (unsaved)
   - Create a new graph
   - `FSGraph > Tools > Explore graph paths`
   - Click on run
   - Error message because we explore an unsaved graph (expected)

3. New graph (saved)
   - Create a new graph
   - Save it
   - `FSGraph > Tools > Explore graph paths`
   - Click on run
   - We have the expected output

4. New graph
   - Create a new graph
   - `FSGraph > Tools > Explore graph paths`
   - Click on run
   - Error message because we explore an unsaved graph (expected)
   - Go back to the graph frame
   - Save the graph
   - Go back to the graph path frame
   - Give an output file name
   - Click on run
   - We have the expected output

5. Several graphs (same directory)
   - Open several graphs, from the same directory
   - `FSGraph > Tools > Explore graph paths`
   - All graphs are mentioned in the ComboBox
   - Choose a graph and click on run
   - We have the expected output
  
6. Several graphs (different directory)
   - Open several graphs, from  different directory
   - `FSGraph > Tools > Explore graph paths`
   - All graphs are mentioned in the ComboBox
   - Choose a graph and click on run
   - We have the expected output
  
7. Reactive frame
   - Open several graphs
   - `FSGraph > Tools > Explore graph paths`
   - Close all graphs frames
   - The GraphPathFrame is automatically closed (expected)

8. Access to `FSGraph > Tools > Explore graph paths` even if the focused frame is not a graph frame
   - Open a graph
   - Open another frame, a DELA dictionary for example
   - `FSGraph > Tools > Explore graph paths` is available (expected)

## Type of files

- [ ] `bin`: Binary files
- [ ] `ci`: Continuous integration files
- [ ] `doc`: Documentation files
- [x] Plain-text source code files

## Level of change

- [ ] `break`: Breaking change
- [ ] `exp`: Experimental change
- [ ] `tmp`: Temporal change
- [ ] `major`: Major change
- [x] `minor`: Minor change
- [ ] `revert`: Revert a commit change
- [ ] `sec`: Vulnerability-related change
- [ ] None of the above (normal change)

## Type of change

- [ ] `deprecat`: Deprecation of a once-stable feature
- [ ] `enhance`: Enhancement in existing functionality
- [ ] `hotfix`: Hotfix for bugs
- [ ] `refactor`: Improve coding style, comments
- [ ] `remove`: Remove a feature

## Checklist:


- [x] My code compiles
- [x] My code does not generate new warnings
- [x] My code only implements a single fix or feature
- [ ] My code does not contain any changes unrelated to the fix or feature
- [x] My code includes javadoc/doxygen where appropriate
- [x] My code is well factored, so that there is not repetitive code in the wild
- [x] My code does not refactor the surrounding code unless necessary
- [x] My code does not require a change in the documentation, if so I already opened an issue to list the changes
- [x] I have read the [**CONTRIBUTING**](../.github/CONTRIBUTING.md) document
- [x] I have read the [**Pull Request/Commit Message Guidelines**](https://git.io/JuS6u)
- [x] I have given a clear and concise title to my pull request following the above guidelines
- [x] I understand that all commits on my pull request will be squashed to a single good one
- [ ] All above points were checked and are marked
